### PR TITLE
[risk=low][RW-6432] Add a dummy CDR Version to Local

### PR DESCRIPTION
**Pull request by jmthibault79**
Will add the other environments before merging.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/cdr/CdrDbConfig.java [hunk 1]
```diff
@@ -46,29 +41,18 @@
...
-    // See also https://precisionmedicineinitiative.atlassian.net/browse/RW-6432
-
-    private final Long defaultCdrVersionId;
-
     @Autowired
     public CdrDataSource(
         CdrVersionDao cdrVersionDao,
         @Qualifier("poolConfiguration") PoolConfiguration basePoolConfig,
-        @Qualifier("cdrPoolConfiguration") PoolConfiguration cdrPoolConfig,
-        @Qualifier("configCache") LoadingCache<String, Object> configCache)
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/cdr/CdrDbConfig.java:69]: configCache and workbenchConfig are not used

File: api/config/cdr_config_local.json [hunk 2]
```diff
@@ -22,6 +22,20 @@
     }
   ],
   "cdrVersions": [
+    {
+      "cdrVersionId": -1,
+      "name": "Dummy version for Spring initialization - do not use",
```
Comment (calbach) [inline api/config/cdr_config_local.json:27]: I don't mind the idea of having a dummy version server side, if that's the cleanest solution - but I dislike putting this into the config; feels like an unnecessary leaking of abstractions.

If our Spring initialization can't run without a dummy value, it feels like the onus should be on the Spring initialization code to solve the problem. Can you comment a bit on where the sticking points are here? Lets say we instead:

1. Just threw an error instead, when the default CDR is requested
2. or, just returned null if no CDR version context is configured

What breaks?

Alternatively:

3. Creating a stub CdrVersion object on the server (that never gets stored / saved)

Comment (jmthibault79) [reply api/config/cdr_config_local.json:27]: 1 is essentially https://github.com/all-of-us/workbench/pull/6231 - I see NPEs and stack traces on dev-up.

**Comment by jmthibault79** [comment]
Closing in favor of #6231

**Pull request closed without merging**